techbuzz/src/main/java/com/sivalabs/techbuzz/posts/domain/services/PostService.java

```diff
@@ -49,6 +49,12 @@ public Post getPost(Long postId) {
         return postRepository.findById(postId).orElseThrow(() -> new ResourceNotFoundException("Post not found"));
     }
 
+    public PostViewDTO getPostViewDTO(Long postId) {
+        log.debug("Fetching post by id: {}", postId);
+        Post post = postRepository.findById(postId).orElseThrow(() -> new ResourceNotFoundException("Post not found"));
```

Comment:
Instead of repeating the logic, we can call `Post post = this.getPost(postId);`

Reply:
Done